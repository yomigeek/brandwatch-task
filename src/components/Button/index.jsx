import React from "react";
import styled from "styled-components";

const CustomButton = styled.button`
  background: ${(props) => (props.disabled ? "#5586A4" : "#0c3e5d")};
  display: inline-block;
  border-radius: 5px;
  padding: 10px 0;
  margin: 15px;
  width 150px;
  border: none;
  color: #fff;
  font-size: 12px;
  cursor: ${(props) => (props.disabled ? "default" : "pointer")};
`;

const Button = ({ title, actionHandler, disabled = false }) => {
  return (
    <CustomButton onClick={actionHandler} disabled={disabled}>
      {title}
    </CustomButton>
  );
};

export default Button;
