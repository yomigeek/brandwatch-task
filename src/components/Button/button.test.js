/* eslint-disable no-undef */
import { render } from '@testing-library/react'
import Button from './index'

test('renders component snapshot', () => {
    render(<Button />)
    const asFragment = render(<Button />).asFragment
    expect(asFragment()).toMatchSnapshot()
})
