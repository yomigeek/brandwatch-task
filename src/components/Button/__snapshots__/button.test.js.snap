// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL dtufTo"
  />
</DocumentFragment>
`;
